core/deployment/src/main/java/io/quarkus/deployment/util/JandexUtil.java

```diff
@@ -374,7 +374,7 @@ public static String getBoxedTypeName(Type type) {
                         return "java.lang.Short";
                 }
         }
-        return type.toString();
+        return type.name().toString();
```

Comment:
That one actually should stay as is. IIRC, the method relies on the behavior of `type.toString()` for some cases and it was one of the reasons why we changed this behavior and deprecated this method (it was causing issues in some cases and actually making things work in others). This method being deprecated since 2.13, maybe we should just remove it for 2.15.

Reply:
Let's see what @Ladicek has to say here

Reply:
Here is the PR where we deprecated the method: https://github.com/quarkusio/quarkus/pull/28536 .

Reply:
I'm more than happy to remove it if @Ladicek also agrees

Reply:
I absolutely support removing this method. I don't think it does anything useful and Quarkus core (the main repo) doesn't seem to use it at all. (Same method actually exists in RESTEasy Reactive, but that doesn't seem to use it either.)

Reply:
It's now gone